sklearn/cluster/tests/test_k_means.py

```diff
@@ -53,11 +53,19 @@
 X, true_labels = make_blobs(
     n_samples=n_samples, centers=centers, cluster_std=1.0, random_state=42
 )
-X_csr = sp.csr_matrix(X)
+X_as_any_csr = [container(X) for container in CSR_CONTAINERS]
+data_containers = [np.array] + CSR_CONTAINERS
+data_containers_ids = (
+    ["dense", "sparse_matrix", "sparse_array"]
+    if len(X_as_any_csr) == 2
+    else ["dense", "sparse_matrix"]
+)
 
 
 @pytest.mark.parametrize(
-    "array_constr", [np.array, sp.csr_matrix], ids=["dense", "sparse"]
+    "array_constr",
+    data_containers,
+    ids=data_containers_ids,
 )
```

Comment:
For all these new parametrization, you can make it fit on a single line:

```python
@pytest.mark.parametrize("array_constr", data_containers, ids=data_containers_ids)
```

It will be a more compact diff. You can make the changes below as well.